core/src/main/java/org/testcontainers/dockerclient/DockerClientProviderStrategy.java

```diff
@@ -104,6 +121,71 @@ public DockerClient getClient() {
         return dockerClient;
     }
 
+    /**
+     * TODO we should consider moving this to docker-java at some point
+     */
+    @UnstableAPI
+    protected boolean test() {
+        TransportConfig transportConfig = getTransportConfig();
+        URI dockerHost = transportConfig.getDockerHost();
+
+        Callable<Socket> socketProvider;
+        SocketAddress socketAddress;
+        switch (dockerHost.getScheme()) {
+            case "tcp":
+            case "http":
+            case "https":
+                SocketFactory socketFactory = SocketFactory.getDefault();
+                SSLConfig sslConfig = transportConfig.getSslConfig();
+                if (sslConfig != null) {
+                    try {
+                        socketFactory = sslConfig.getSSLContext().getSocketFactory();
+                    } catch (KeyManagementException | UnrecoverableKeyException | NoSuchAlgorithmException | KeyStoreException e) {
+                        log.warn("Exception while creating SSLSocketFactory", e);
+                        return false;
+                    }
+                }
+                socketProvider = socketFactory::createSocket;
+                socketAddress = new InetSocketAddress(dockerHost.getHost(), dockerHost.getPort());
+                break;
+            case "unix":
+            case "npipe":
+                if (!new File(dockerHost.getPath()).exists()) {
+                    log.debug("DOCKER_HOST socket file '{}' does not exist", dockerHost.getPath());
```

Comment:
```suggestion
                    log.warn("DOCKER_HOST socket file '{}' does not exist", dockerHost.getPath());
```

Isn't this an unexpected case?

Reply:
it is not, just the file is missing, so we skip the strategy, nothing exceptional about this situation (we expect this to happen)